ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/SingleEntityVulnerabilitiesTable.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { Button, ButtonVariant } from '@patternfly/react-core';
+import { TableComposable, Thead, Tr, Th, Tbody, Td } from '@patternfly/react-table';
+import { CheckCircleIcon, ExclamationCircleIcon } from '@patternfly/react-icons';
+import { SVGIconProps } from '@patternfly/react-icons/dist/js/createIcon';
+
+import LinkShim from 'Components/PatternFly/LinkShim';
+import SeverityIcons from 'Components/PatternFly/SeverityIcons';
+import { vulnerabilitySeverityLabels } from 'messages/common';
+import { getDistanceStrictAsPhrase } from 'utils/dateUtils';
+import { ImageVulnerabilitiesResponse } from './hooks/useImageVulnerabilities';
+import { getEntityPagePath } from './searchUtils';
+
+export type SingleEntityVulnerabilitiesTableProps = {
+    // TODO - Note that we may need to adjust this prop type, or do some pre-processing of data
+    //        when this component is used for the Deployment single page
+    vulnerabilities: ImageVulnerabilitiesResponse['image']['imageVulnerabilities'];
+};
+
+function SingleEntityVulnerabilitiesTable({
+    vulnerabilities,
+}: SingleEntityVulnerabilitiesTableProps) {
+    return (
+        <TableComposable>
+            <Thead>
+                <Tr>
+                    <Th>CVE</Th>
+                    <Th>Severity</Th>
+                    <Th>CVE Status</Th>
+                    <Th>Affected components</Th>
+                    <Th>First discovered</Th>
+                </Tr>
+            </Thead>
+            <Tbody>
+                {vulnerabilities.map(
+                    ({ cve, severity, isFixable, imageComponents, discoveredAtImage }) => {
+                        const SeverityIcon: React.FC<SVGIconProps> | undefined =
```

Comment:
Two reasons for the explicit type annotations:

1. The backend is returning a `string` for `severity` here, instead of a GraphQL enum for the severity types. Directly accessing `SeverityIcons[severity]` silently returns a type of `any` instead of `React.FC`.
2. After investigating the above, I found it is _technically_ possible for the backend to return a severity of `UNKNOWN_VULNERABILITY_SEVERITY`, which we don't handle on the front end, so it is possible that this component will be `undefined` and crash the UI if the "unknown" value makes it through somehow.